neard/src/runtime/mod.rs

```diff
@@ -1863,44 +1854,26 @@ mod test {
         pub fn compute_reward(
             &self,
             num_validators: usize,
-            #[cfg(feature = "protocol_feature_rectify_inflation")] epoch_duration: u64,
+            epoch_duration: u64,
         ) -> (Balance, Balance) {
-            #[cfg(not(feature = "protocol_feature_rectify_inflation"))]
-            {
-                let per_epoch_total_reward = *self.runtime.genesis_config.max_inflation_rate.numer()
-                    as u128
-                    * self.runtime.genesis_config.total_supply
-                    * self.runtime.genesis_config.epoch_length as u128
-                    / (self.runtime.genesis_config.num_blocks_per_year as u128
-                        * *self.runtime.genesis_config.max_inflation_rate.denom() as u128);
-                let per_epoch_protocol_treasury = per_epoch_total_reward
-                    * *self.runtime.genesis_config.protocol_reward_rate.numer() as u128
-                    / *self.runtime.genesis_config.protocol_reward_rate.denom() as u128;
-                let per_epoch_per_validator_reward =
-                    (per_epoch_total_reward - per_epoch_protocol_treasury) / num_validators as u128;
-                (per_epoch_per_validator_reward, per_epoch_protocol_treasury)
-            }
-            #[cfg(feature = "protocol_feature_rectify_inflation")]
-            {
-                let num_seconds_per_year = 60 * 60 * 24 * 365;
-                let num_ns_in_second = 1_000_000_000;
-                let per_epoch_total_reward =
-                    (U256::from(*self.runtime.genesis_config.max_inflation_rate.numer() as u64)
-                        * U256::from(self.runtime.genesis_config.total_supply)
-                        * U256::from(epoch_duration)
-                        / (U256::from(num_seconds_per_year)
-                            * U256::from(
-                                *self.runtime.genesis_config.max_inflation_rate.denom() as u128
-                            )
-                            * U256::from(num_ns_in_second)))
-                    .as_u128();
-                let per_epoch_protocol_treasury = per_epoch_total_reward
-                    * *self.runtime.genesis_config.protocol_reward_rate.numer() as u128
-                    / *self.runtime.genesis_config.protocol_reward_rate.denom() as u128;
-                let per_epoch_per_validator_reward =
-                    (per_epoch_total_reward - per_epoch_protocol_treasury) / num_validators as u128;
-                (per_epoch_per_validator_reward, per_epoch_protocol_treasury)
-            }
+            let num_seconds_per_year = 60 * 60 * 24 * 365;
+            let num_ns_in_second = 1_000_000_000;
+            let per_epoch_total_reward =
+                (U256::from(*self.runtime.genesis_config.max_inflation_rate.numer() as u64)
```

Comment:
Maybe use `Ratio`, if computation shall be precise?

Reply:
How does it make it more precise? This number must be an integer

Reply:
You use integer division with multiple division and multiplication, Ratio can try make intermediate steps more precise.